homeassistant/components/shelly/__init__.py

```diff
@@ -142,6 +146,60 @@ async def async_device_setup(
         )
 
 
+async def async_services_setup(
+    hass: HomeAssistant, dev_reg: device_registry.DeviceRegistry
+):
+    """Set up services."""
+
+    async def async_service_ota_update(call: ServiceCall):
+        """Trigger OTA update."""
+        if not (call.data.get(ATTR_DEVICE_ID) or call.data.get(ATTR_AREA_ID)):
+            _LOGGER.warning("OTA update service: no device or area selected")
+            return
+
+        devices = []
+        if call.data.get(ATTR_AREA_ID):
+            for area_id in call.data.get(ATTR_AREA_ID):
+                devices += [
+                    area_dev
+                    for area_dev in device_registry.async_entries_for_area(
+                        dev_reg, area_id
+                    )
+                    if DOMAIN in next(iter(area_dev.identifiers))
+                ]
+
+        if call.data.get(ATTR_DEVICE_ID):
+            for device_id in call.data.get(ATTR_DEVICE_ID):
+                device = dev_reg.async_get(device_id)
+                if not any(device.id == x.id for x in devices):
+                    devices += [device]
+
+        for device in devices:
+            entry_id = next(iter(device.config_entries))
+            entry_data = hass.data[DOMAIN][DATA_CONFIG_ENTRY][entry_id]
+            rest: ShellyDeviceRestWrapper = entry_data[REST]
+            update_data = rest.device.status["update"]
+
+            if not update_data["has_update"]:
+                _LOGGER.info("No OTA updates for %s available", device.name)
+                continue
+
+            if update_data["status"] == "updating":
+                _LOGGER.warning("OTA update already in progress for %s", device.name)
+                continue
+
+            _LOGGER.debug(
+                "OTA update service - trigger OTA update for device %s from '%s' to '%s'",
+                device.name,
+                update_data["old_version"],
+                update_data["new_version"],
+            )
+            resp = await rest.device.http_request("get", "ota", {"update": "true"})
```

Comment:
Also here, need to wrap with timeout and catch the error.

**Note:** We intentionally don't do any http request directly from HA, but add a method in `aioshelly` library and call it from HA, might be a better option to do the same here

Reply:
yeah for sure ... I've created an PR home-assistant-libs/aioshelly#90 to add a new device method ... I will use it later in this PR